py3status/modules/weather_owm.py

```diff
@@ -427,7 +428,13 @@ def post_config_hook(self):
 
     def _make_req(self, url, params=None):
         # Make a request expecting a JSON response
-        req = self.py3.request(url, params=params)
+        try:
+            req = self.py3.request(url, params=params)
+        # retry every 10 seconds when the issue is caused by network
+        # (e. g. temporary issue with resolving)
+        except (RequestURLError, RequestTimeout):
+            return self.py3.error('error', timeout=10)
```

Comment:
I prefer this.
1) Not importing them again.
2) Display error messages instead of static string "error".
```diff
diff --git a/py3status/modules/weather_owm.py b/py3status/modules/weather_owm.py
index 0db3e9e..8cb3ef1 100644
--- a/py3status/modules/weather_owm.py
+++ b/py3status/modules/weather_owm.py
@@ -261,7 +261,6 @@ diff
 """
 
 import datetime
-from py3status.exceptions import RequestTimeout, RequestURLError
 
 
 # API information
@@ -432,8 +431,8 @@ class Py3status:
             req = self.py3.request(url, params=params)
         # retry every 10 seconds when the issue is caused by network
         # (e. g. temporary issue with resolving)
-        except (RequestURLError, RequestTimeout):
-            return self.py3.error('error', timeout=10)
+        except (self.py3.RequestURLError, self.py3.RequestTimeout) as err:
+            return self.py3.error(str(err), timeout=10)
 
         if req.status_code != 200:
             data = req.json()
```
Last thing left is to see if we want hardcoded `timeout=10` or to make `error_timeout` possible similar to `request_timeout` or we could wrongly apply `request_timeout` to this instead. Which one? @ultrabug

Reply:
> I prefer this.
> 
>     1. Not importing them again.

without import of the exceptions I get `NameError`.

Reply:
Is that with `self.py3.RequestURLError, self.py3.RequestTimeout`? I added `self.py3.`

Reply:
true, thank you!

Reply:
ultrabug didn't get back to us. Can you rename `request_timeout` to `retry_timeout` (This name is borrowed from deprecated module `weather_yahoo` with same purpose). This also allow `request_timeout` to work correctly (i.e. separately)... to prevent any grief... then yeah, I guess we can merge it asap.